gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,383 @@
+ * starting with the current settings for their applications on the local machine.
+ *
+ * @param {lifecycleManager.invokeSettingsHandlerGet} invokeSettingsHandlerGet - Invoker from the `lifecycleManager`
+ * necessary for extracting settings from a solution and it's spec.
+ * @param {lifecycleManager.variableResolver} resolver - Resolver for looking up handler specifications
+ * @param {lifecycleManager.userSession.localResolver} localResolver - Resolver for expanding configuration file paths
+ * from the solutions registry.
+ * @param {Object} solutions - Solutions registry entries for solutions available on the current machine.
+ * @param {Object} options - Extra options for processing.
+ * @param {Array} options.solutionsList - If provided, only solutions in this list of `solutionsID`s will
+ * be captures. Example:
+ *
+ * '''json
+ * ["com.microsoft.windows.cursors", "com.freedomscientific.jaws"]
+ * '''
+ * @return {fluid.promise} Returns a promise resolving with the entire system settings capture.
+ */
+gpii.flowManager.captureSystemSettings = function (invokeSettingsHandlerGet, resolver, localResolver, solutions, options) {
+    var promiseTogo = fluid.promise();
+
+    var handlerGetPromises = [];
+
+    var safeHandlerGet = function (invokeSettingsHandlerGet, solutionID, handlerSpec) {
+        var promiseTogo = fluid.promise();
+        try {
+            invokeSettingsHandlerGet(solutionID, handlerSpec).then(function (data) {
+                // Fix up the nativeWindows and SPI Settings Handlers
+                var filteredData = gpii.flowManager.filterInvokeSettingsHandlersReturn(data, handlerSpec);
+                promiseTogo.resolve(filteredData);
+            }, function (err) {
+                promiseTogo.resolve({
+                    solutionId: solutionID,
+                    handlerSpec: handlerSpec,
+                    isError: true,
+                    message: "inner promise error",
+                    err: err
+                });
+            });
+        }
+        catch (err) {
+            promiseTogo.resolve({
+                solutionId: solutionID,
+                handlerSpec: handlerSpec,
+                isError: true,
+                message: "outer try error",
+                err: err
+            });
+        }
+        return promiseTogo;
+    };
+
+    var solutionsToFetch = solutions;
+    if (options.solutionsList) {
+        solutionsToFetch = {};
+        fluid.each(solutions, function (solution, solutionID) {
+            if (options.solutionsList.indexOf(solutionID) >= 0) {
+                solutionsToFetch[solutionID] = solution;
+            }
+        });
+    }
+
+    fluid.each(solutionsToFetch, function (solution, solutionID) {
+        var newSolution = gpii.lifecycleManager.transformSolutionSettings(solution, gpii.settingsHandlers.changesToSettings);
+
+        if (solutionID === "com.freedomscientific.jaws") {
```

Comment:
I'm worried there are so many special handlings hardcoded specific windows solutions in this branch.

@amb26, any suggestions?

Reply:
Yes, all of this code needs to go out into the Windows repository and get invoked by the settings handlers. Ideally it should go into a separate file named after JAWS, if we get a lot of such hacks we should start thinking of how to encode the fact we need to invoke them in the solutions registry. But at the very least, this stuff needs to get out of universal.

Reply:
These have been removed from the universal branch

Reply:
removed from universal